File: test/static/lang/java/GenericExceptionsClose.java
import java.security.SecureRandom;
import java.security.MessageDigest;
import java.security.NoSuchAlgorithmException;

class GenericExceptionsClose{
   public static void main(String args[]){
     try{
         int a[]=new int[7];
         SecureRandom random = new SecureRandom();
         byte bytes[] = new byte[20];
         a[0]=random.nextBytes(bytes);
         a[4]=30/0;
         System.out.println("First print statement in try block");

         MessageDigest messageDigest, messageDigest2;
         messageDigest = MessageDigest.getInstance("SHA-256");
         messageDigest.update(data.getBytes());
         byte[] messageDigestSHA256 = messageDigest.digest();

         Cipher aes = Cipher.getInstance("AES");
         aes.init(Cipher.ENCRYPT_MODE, secretKeySpec);
         byte[] encrypted = aes.doFinal(input.getBytes("UTF-8"));

         if (a[0] > 200) {
            System.out.println("Big num");
         } else  if (a[0] < 100){
            System.out.println("Small num");
         } else {
            System.out.println("Average num");
         }
     }
     catch(ArithmeticException e){
        System.out.println("Warning: ArithmeticException");
     }
     catch(ArrayIndexOutOfBoundsException e){
        System.out.println("Warning: ArrayIndexOutOfBoundsException");
     }
     catch (NoSuchAlgorithmException exception) {
        System.out.println("Warning: NoSuchAlgorithmException");
     }
/*
     try {
           System.out.println("Out of try-catch block...");
           int a = Math.random();
           if (a[0] > 200) {
              System.out.println("Big num");
           }
  catch(Exception e){
        System.out.println("Warning: Some Other exception");
     }
*/
  }
}

File: test/static/lang/csharp/GenericExceptionsClose.cs
using System;
using System.Security.Cryptography;

public class GenericExceptionsClose
{
  public static void Main()
  {
    try
    {
      int[] a = new int[7];
      a[4]=30/0;
      Console.WriteLine("First print statement in try block");
      if (a > 5) {
        Console.WriteLine("Big num")
      } else {
        Console.WriteLine("Small num")
      }
    }

    catch(ArithmeticException e)
    {
      Console.WriteLine("Warning: ArithmeticException");
    }

    catch(IndexOutOfRangeException e)
    {
      Console.WriteLine("Warning: ArrayIndexOutOfBoundsException");
    }
/*
    try
    {
      Console.WriteLine("Out of try-catch block...");
      if (a > 5) {
        Debug.WriteLine("Big num")
      }
      MD5 md5 = System.Security.Cryptography.MD5.Create();
      byte[] inputBytes = System.Text.Encoding.ASCII.GetBytes(input);
      byte[] hash = md5.ComputeHash(inputBytes);

      MD5CryptoServiceProvider md5Hasher = new MD5CryptoServiceProvider();
      byte[] data = md5Hasher.ComputeHash(Encoding.Default.GetBytes(input));

      SHA1 shaM = new SHA1Managed();
      result = shaM.ComputeHash(data);
    }

    catch(Exception e)
    {
      Console.WriteLine("Warning: Some Other exception");
    }
*/
  }
}
